tests/common/snappi/traffic_generation.py

```diff
@@ -0,0 +1,571 @@
+
+    return base_flow_config
+
+
+def generate_test_flows(testbed_config,
+                        test_flow_name,
+                        test_flow_prio_list,
+                        test_flow_rate_percent,
+                        data_flow_dur_sec,
+                        data_flow_delay_sec,
+                        data_pkt_size,
+                        prio_dscp_map,
+                        snappi_extra_params):
+    """
+    Generate test configurations of flows. Test flows and background flows are also known as data flows.
+
+    Args:
+        testbed_config (obj): testbed L1/L2/L3 configuration
+        test_flow_name (str): name of test flow
+        test_flow_prio_list (list): list of test flow priorities
+        test_flow_rate_percent (int): rate percentage of test flows
+        data_flow_dur_sec (int): duration of test flows
+        data_flow_delay_sec (int): delay of test flows in seconds
+        data_pkt_size (int): packet size of test flows
+        prio_dscp_map (dict): priority to DSCP mapping
+        snappi_extra_params (obj): snappi extra parameters
+    """
+    base_flow_config = snappi_extra_params.base_flow_config
+    pytest_assert(base_flow_config is not None, "Cannot find base flow configuration")
+
+    for prio in test_flow_prio_list:
+        test_flow = testbed_config.flows.flow(name='{} Prio {}'.format(test_flow_name, prio))[-1]
+        test_flow.tx_rx.port.tx_name = base_flow_config["tx_port_name"]
+        test_flow.tx_rx.port.rx_name = base_flow_config["rx_port_name"]
+
+        eth, ipv4 = test_flow.packet.ethernet().ipv4()
+        eth.src.value = base_flow_config["tx_mac"]
+        eth.dst.value = base_flow_config["rx_mac"]
+        eth.pfc_queue.value = prio
+
+        ipv4.src.value = base_flow_config["tx_port_config"].ip
+        ipv4.dst.value = base_flow_config["rx_port_config"].ip
+        ipv4.priority.choice = ipv4.priority.DSCP
+        ipv4.priority.dscp.phb.values = prio_dscp_map[prio]
+        ipv4.priority.dscp.ecn.value = (
+            ipv4.priority.dscp.ecn.CAPABLE_TRANSPORT_1)
+
+        test_flow.size.fixed = data_pkt_size
+        test_flow.rate.percentage = test_flow_rate_percent
+        test_flow.duration.fixed_seconds.seconds = data_flow_dur_sec
+        test_flow.duration.fixed_seconds.delay.nanoseconds = int(sec_to_nanosec(data_flow_delay_sec))
+
+        test_flow.metrics.enable = True
+        test_flow.metrics.loss = True
+
+        """ Set flow port config values """
+        flow_port_config = base_flow_config["flow_port_config"]
+        flow_port_config[0][str(base_flow_config["tx_port_config"].peer_port)].append(int(prio))
+        flow_port_config[1][str(base_flow_config["rx_port_config"].peer_port)].append(int(prio))
+        base_flow_config["flow_port_config"] = flow_port_config
+
+    snappi_extra_params.base_flow_config = base_flow_config
+
+
+def generate_background_flows(testbed_config,
```

Comment:
Can we use a single function to generate both test flows and background flows?

Reply:
I am keeping them separate in case in the future we want to only run test flows or background flows. I would also want one traffic gen function to generate exactly one type of flow.